### PR TITLE
Improvement in VideoRipper

### PR DESCRIPTION
**Pull request by Gaboso**
Declared 'bytesCompleted' on a separate line.
'canRip' was defined in the 'RipperInterface' interface and therefore was removed.
Redundant variable removed.
Closed fileWritter.

### CODE REVIEW

File: src/main/java/com/rarchives/ripme/ripper/VideoRipper.java [hunk 1]
```diff
@@ -148,11 +156,25 @@ public URL sanitizeURL(URL url) throws MalformedURLException {
      */
     @Override
     protected void checkIfComplete() {
-        if (observer == null) {
+        if (observer == null)
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/VideoRipper.java]: keep braces

File: src/main/java/com/rarchives/ripme/ripper/VideoRipper.java [hunk 2]
```diff
@@ -109,33 +117,33 @@ public void downloadCompleted(URL url, File saveAs) {
...
     public String getStatusText() {
-        StringBuilder sb = new StringBuilder();
-        sb.append(getCompletionPercentage())
-          .append("% ")
-          .append(" - ")
-          .append(Utils.bytesToHumanReadable(bytesCompleted))
-          .append(" / ")
-          .append(Utils.bytesToHumanReadable(bytesTotal));
-        return sb.toString();
+        return String.valueOf(getCompletionPercentage()) +
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/VideoRipper.java]: Raw String concatenation is slower than StringBuilder. Why did you make this change?

File: src/main/java/com/rarchives/ripme/ripper/VideoRipper.java [hunk 3]
```diff
@@ -109,33 +117,33 @@ public void downloadCompleted(URL url, File saveAs) {
...
-        }
+
         observer.update(this, new RipStatusMessage(STATUS.DOWNLOAD_ERRORED, url + " : " + reason));
         checkIfComplete();
     }
+
     @Override
     public void downloadExists(URL url, File file) {
-        if (observer == null) {
+        if (observer == null)
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/VideoRipper.java]: keep braces throughout your refactorings, please

File: src/main/java/com/rarchives/ripme/ripper/VideoRipper.java [hunk 4]
```diff
@@ -44,19 +47,22 @@ public boolean addURLToDownload(URL url, File saveAs) {
...
                 fw.write("\n");
-                fw.close();
+
                 RipStatusMessage msg = new RipStatusMessage(STATUS.DOWNLOAD_COMPLETE, urlFile);
                 observer.update(this, msg);
             } catch (IOException e) {
                 logger.error("Error while writing to " + urlFile, e);
                 return false;
+            } finally {
+                close(fw);
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/VideoRipper.java]: I see why you did this, but maybe there's a slight change to the pattern that would allow you to do fw.close() in the finally block...?

File: src/main/java/com/rarchives/ripme/ripper/VideoRipper.java [hunk 5]
```diff
@@ -1,34 +1,37 @@
...
-import com.rarchives.ripme.ui.RipStatusMessage;
-import com.rarchives.ripme.ui.RipStatusMessage.STATUS;
-import com.rarchives.ripme.utils.Utils;
-
 public abstract class VideoRipper extends AbstractRipper {
 
-    private int bytesTotal = 1,
-                 bytesCompleted = 1;
+    private int bytesTotal = 1;
+    private int bytesCompleted = 1;
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/VideoRipper.java]: Thank you for this. This really gets on my nerves :)

**Review by metaprime — CHANGES_REQUESTED** [review]

File: src/main/java/com/rarchives/ripme/ripper/VideoRipper.java (hunk 2 above)
Comment (Gaboso) [inline src/main/java/com/rarchives/ripme/ripper/VideoRipper.java]: I had thought of making it simpler and shorter, but I did not analyze the difference between the processing speed of both

File: src/main/java/com/rarchives/ripme/ripper/VideoRipper.java (hunk 4 above)
Comment (Gaboso) [inline src/main/java/com/rarchives/ripme/ripper/VideoRipper.java]: If i add fw.close in the finally block, i will have to add or try/cacth around it, but i believe that try/catch inside the finally block is bad practice, so i created the close method, which contains the necessary try/catch to run fw.close

File: src/main/java/com/rarchives/ripme/ripper/VideoRipper.java (hunk 2 above)
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/VideoRipper.java]: I'd sooner keep it as-is because someone put thought into performance here before. It's probably not a big difference, but no need to undo the good work.

Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/VideoRipper.java]: I acknowledge there may actually be no difference because of compiler optimizations, but until I have a chance to look into it, I don't want to make this call just yet. https://stackoverflow.com/questions/1532461/stringbuilder-vs-string-concatenation-in-tostring-in-java

File: src/main/java/com/rarchives/ripme/ripper/VideoRipper.java (hunk 4 above)
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/VideoRipper.java]: Seems fine but I'm wondering whether there is a well-established pattern to solve this problem already that might look a little more elegant. Considering this is a fundamental problem of classes like this, I would think there is something.

No need to make any changes here, just wondering.

File: src/main/java/com/rarchives/ripme/ripper/VideoRipper.java (hunk 2 above)
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/VideoRipper.java]: Would prefer you to undo this change before I merge.

File: src/main/java/com/rarchives/ripme/ripper/VideoRipper.java [hunk 6]
```diff
@@ -70,22 +76,24 @@ public boolean addURLToDownload(URL url, File saveAs) {
...
         return addURLToDownload(url, saveAs);
     }
 
     @Override
     public void setWorkingDir(URL url) throws IOException {
         String path = Utils.getWorkingDirectory().getCanonicalPath();
-        if (!path.endsWith(File.separator)) {
+        if (!path.endsWith(File.separator))
             path += File.separator;
-        }
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/VideoRipper.java]: Looks like you missed a pair of braces.

File: src/main/java/com/rarchives/ripme/ripper/VideoRipper.java [hunk 7]
```diff
@@ -96,9 +104,9 @@ public int getCompletionPercentage() {
 
     @Override
     public void downloadCompleted(URL url, File saveAs) {
-        if (observer == null) {
+        if (observer == null)
             return;
-        }
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/VideoRipper.java]: Another pair of braces.

File: src/main/java/com/rarchives/ripme/ripper/VideoRipper.java [hunk 8]
```diff
@@ -109,14 +117,16 @@ public void downloadCompleted(URL url, File saveAs) {
             logger.error("Exception while updating observer: ", e);
         }
     }
+
     @Override
     public void downloadErrored(URL url, String reason) {
-        if (observer == null) {
+        if (observer == null)
             return;
-        }
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/VideoRipper.java]: Another

**Review by metaprime — CHANGES_REQUESTED** [review]

**Comment by metaprime** [comment]
@Gaboso Why did you close the PR? Apologies for not following up on this.
